rfcs/0000-central-station.md

```diff
@@ -0,0 +1,507 @@
+is at the final terminal, and so on. In addition, in the cases when we get this wrong,
+CentralStation receives API events from KVL, Glides, and Skate, updating the tag.
+
+Furthermore, CentralStation manages the "Service Changes" to trips. If Keolis, Glides, or Skate
+updates CentralStation that a new trip is being created, a trip is being canceled, or the trip's
+properties (stops, shape, headsign, etc) are being changed, CentralStation stores this information
+and uses it to produce GTFS ServiceChanges files. Swiftly and RTR statelessly consume these
+VehiclePositions and ServiceChanges to produce predictions. (This relies on us using our combined
+mode leverage and Swiftly's stated interest in consuming a feed like this; currently they only
+offer a proprietary API. If they don't get around to consuming a feed, CentralStation will instead
+use their API, tying us to Swiftly, but only at the CentralStation boundary, and not at Glides,
+Skate, Keolis levels.)
+
+CentralStation periodically produces the public GTFS VehiclePositions feed consumed by
+concentrate, by selecting its revenue vehicles and public fields. In the future, it may produce
+the Alerts and TripUpdates fields, too, using Swiftly and/or RTR and/or TransitClock as
+implementation details, and applying alerts and other business logic on top. But for now, we lean
+on the existing alert-applying logic in concentrate, and expect RTR and Swiftly to provide the
+TripUpdates to concentrate.
+
+If internal CTD apps need data not in the public feed, they can access it via CentralStation's
+API. For example, Glides will use CentralStation to access all the non-revenue vehicles, and Skate
+can get block and run information. (And operator info, like name? I don't know if PI info like
+that needs to live in CentralStation.)
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+Here are the effects of this proposal on the various pieces of our current realtime data
+ecosystem, starting with the idea of Service Changes, then all our existing apps, and finally
+CentralStation itself.
+
+## Service Changes
+
+A major component of this proposal, and a strong focus of CTD going forward, is the idea of
+"Service Changes". This is a [GTFS-rt proposal that has been discussed for 3.5
+years](https://github.com/google/transit/issues/113). Here is the [most recent version of the
+proposal](https://docs.google.com/document/d/1oPfQ6Xvui0g3xiy1pNiyu5cWFLhvrTVyvYsMVKGyGWM/edit#heading=h.mm6bblup6h8d).
+I refer to these in the RFC as a "ServiceChanges" file, but really the proposal is about new
+FeedEntity enum values, as well as extending the existing TripUpdates one. By convention, we
+produce `VehiclePositions` and `TripUpdates` files separately, though this is not required by the
+spec. We could similarly produce a `ServiceChanges` file composed of feed entities related to
```

Comment:
Yes, I think a separate feed is the way to go here. The way I think about it is how frequently the data is updated:
- VehiclePositions: every second
- TripUpdates: every 10 - 30 seconds
- ServiceChanges: every 1 - 15 minutes